crates/pg_workspace_new/src/workspace/server.rs

```diff
@@ -291,6 +300,50 @@ impl Workspace for WorkspaceServer {
     fn is_path_ignored(&self, params: IsPathIgnoredParams) -> Result<bool, WorkspaceError> {
         Ok(self.is_ignored(params.pglsp_path.as_path()))
     }
+
+    fn pull_diagnostics(
+        &self,
+        params: super::PullDiagnosticsParams,
+    ) -> Result<super::PullDiagnosticsResult, WorkspaceError> {
+        // get all statements form the requested document and pull diagnostics out of every
+        // sourcece
+        let doc = self
+            .documents
+            .get(&params.path)
+            .ok_or(WorkspaceError::not_found())?;
+
+        let diagnostics: Vec<SDiagnostic> = doc
+            .statement_refs_with_ranges()
+            .iter()
+            .flat_map(|(stmt, r)| {
+                let mut stmt_diagnostics = vec![];
+
+                stmt_diagnostics.extend(self.pg_query.pull_diagnostics(stmt));
```

Comment:
```suggestion
                let mut stmt_diagnostics = self.pg_query.pull_diagnostics(stmt);
```

Reply:
Does this work?

Reply:
yes, but I implemented it like this because that's the place to pull diagnostics from other "sources" eg linter. right now your suggestion is better, but once we integrate the linter or typechecker, there will be another `extend` call right below the one fetching form pg query.

Reply:
gotcha!